common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -209,8 +210,16 @@ final int threadLocalSize() {
 
     public interface Handle<T> extends ObjectPool.Handle<T>  { }
 
+    @SuppressWarnings("unchecked")
     private static final class DefaultHandle<T> implements Handle<T> {
```

Comment:
You can just extends `AtomicInteger` that would give you back `Unsafe` perf for free :)

Reply:
I considered it, but then I loose the field name which is quite awkward. Also, I think the JIT is able to trust the finals inside A*FUs these days, so they're as fast as VarHandles or Atomic*s these days.